src/components/Login.jsx

```diff
@@ -1,7 +1,35 @@
+import { useContext, useState } from "react"
+import { UserContext } from "../contexts/user"
+import { getUsers } from "./api"
+
 function Login(){
+  const { username, setUsername } = useContext(UserContext)
+  const [userInput, setUserInput] = useState("")
+  const [userErr, setUserErr] = useState(false)
+
+  const LoginHandler = (event) =>{
+    event.preventDefault()
+    getUsers()
+    .then((users)=>{
+        for(let i = 0; i < users.length; i++){
+            if(users[i].username === userInput){
+                setUsername(users[i])
+            }
+        }
+        if(username === null){
+            setUserErr(true)
+        }
+    })
+  }
     return(
-        <form>
-            <label>Login</label>
+        <form onSubmit={LoginHandler}>
+            <label>Please login</label><br/>
+            <label>Username: </label>
+            <input type='text' onChange={(event)=>setUserInput(event.target.value)}/>
```

Comment:
I believe this form could be amended to be more controlled. Aka setting the `value` property to the state